src/Transport/Receiving/MessagePump.cs

```diff
@@ -197,26 +204,16 @@ await circuitBreaker.Failure(processErrorEventArgs.Exception, processErrorEventA
                 .ConfigureAwait(false);
         }
 
-
-        public async Task ChangeConcurrency(PushRuntimeSettings newLimitations, CancellationToken cancellationToken = default)
+        public Task ChangeConcurrency(PushRuntimeSettings newLimitations, CancellationToken cancellationToken = default)
         {
             limitations = newLimitations;
-            if (transportSettings.PrefetchCount.HasValue)
-            {
-                // For all users that have set a predefined fixed prefetch count we are adjusting the concurrency
-                // by using what the SDK provides since the prefetch count is always fixed.
-                processor.UpdateConcurrency(limitations.MaxConcurrency);
-            }
-            else
-            {
-                // For all other cases the users is using either the default multiplier or a defined multiplier
-                // that sets the prefetch count in accordance of the maximum concurrency. In those scenarios we cannot
-                // use UpdateConcurrency because that would not adjust the prefetch count to the new desired values
-                // therefore we are stopping and restarting (which also creates a new underlying AMQP link that will have
-                // the new prefetch count settings.
-                await StopReceive(cancellationToken).ConfigureAwait(false);
-                await StartReceive(cancellationToken).ConfigureAwait(false);
-            }
+
+            processor.UpdateConcurrency(limitations.MaxConcurrency);
```

Comment:
Do we need that considering that we set the concurrency in [line 82](https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/761/files#diff-bf785c1b37e39481e89856ea4195ca5c7e21f862b8eb2ffc43b63208a3c582e5R77)?

Reply:
The code that sets the concurrency is the code that takes the value from the endpoint configuration. Let's say a user has `LimitProcessingConcurrencyTo(15)` then it sets the maximum concurrency to `15` in the code you mentioned. When the core rate-limiting kicks in it sets the concurrency to `1` until the failures are gone and then sets the concurrency back to `15` in this example. The code here sets the concurrency of the processor from one value to another as instructed by the core.

Or maybe I'm misunderstanding you.